zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/UserTaskHandler.java

```diff
@@ -140,8 +154,14 @@ public void flush(final TaskEntity entity, final BatchRequest batchRequest) {
 
     final Map<String, Object> updateFields = getUpdatedFields(entity);
 
+    final boolean previousVersionRecord = refersToPreviousVersionRecord(entity.getKey());
+
     batchRequest.upsertWithRouting(
-        indexName, entity.getId(), entity, updateFields, entity.getProcessInstanceId());
+        indexName,
+        previousVersionRecord ? String.valueOf(entity.getKey()) : entity.getId(),
+        entity,
+        updateFields,
+        previousVersionRecord ? String.valueOf(entity.getKey()) : entity.getProcessInstanceId());
```

Comment:
🔧 I would like to have some comments, explaining actually what we are doing here and why.

For example: Based on the key, and the first seen created user task key, we can determine that the key is either created before or after.

If Before it means it was stored before with the user task key as ID (in ES/OS), so we have to use the same to update it.

_Thinking of this, might make sense to split the handler to make the logic separate? (not sure here, just a thought)_

If after it means we have to use the id == flownodeinstancekey as the ID of the document.